packages/library-base/src/commonMain/kotlin/io/realm/kotlin/internal/RealmListInternal.kt

```diff
@@ -113,9 +114,17 @@ internal class ManagedRealmList<E>(
         return operator.set(index, element)
     }
 
-    override fun asFlow(): Flow<ListChange<E>> {
+    override fun asFlow(keyPaths: List<String>?): Flow<ListChange<E>> {
         operator.realmReference.checkClosed()
-        return operator.realmReference.owner.registerObserver(this)
+        val keyPathInfo = keyPaths?.let {
```

Comment:
Hmm. Maybe worth examining the `contract` approach if we have a lot of these. Or maybe just use the existing 
```
Validation.checkType(... ).let { Pair(it.classkey, keyPaths) }
```

Reply:
I used the approach outlined here: https://github.com/realm/realm-kotlin/pull/1547#discussion_r1401890531